### PR TITLE
Add publisher publishing function call in various components

### PR DESCRIPTION
**Pull request by ThaminduDilshan**
### Purpose
Add publisher publishing function call in various components' on click and handle on click functions.

### Checklist
- [x] e2e cypress tests locally verified.
- [ ] Manual test round performed and verified.
- [ ] UX/UI review done on final implementation.
- [ ] Documentation provided. (Add links if there's any)
- [ ] Unit tests provided. (Add links if there's any)
- [ ] Integration tests provided. (Add links if there's any)

### Security checks
- [ ] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines?
- [ ] Ran FindSecurityBugs plugin and verified report?
- [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: apps/console/src/app.tsx [hunk 1]
```diff
@@ -70,9 +70,12 @@ export const App: FunctionComponent<{}> = (): ReactElement => {
     const config: ConfigReducerStateInterface = useSelector((state: AppState) => state.config);
     const allowedScopes: string = useSelector((state: AppState) => state?.auth?.scope);
     const appTitle: string = useSelector((state: AppState) => state?.config?.ui?.appTitle);
+    const UUID: string = useSelector((state: AppState) => state.profile.profileInfo.id);
 
     const [ baseRoutes, setBaseRoutes ] = useState<RouteInterface[]>(getBaseRoutes());
 
+    const eventPublisher = EventPublisher.getInstance();
```
Comment (brionmario) [inline apps/console/src/app.tsx:77]:
```suggestion
    const eventPublisher: EventPublisher = EventPublisher.getInstance();
```

Comment (ThaminduDilshan) [reply apps/console/src/app.tsx]: fixed

**Review by brionmario — APPROVED** [review]

File: apps/console/src/app.tsx [hunk 2]
```diff
@@ -70,9 +70,12 @@ export const App: FunctionComponent<{}> = (): ReactElement => {
     const config: ConfigReducerStateInterface = useSelector((state: AppState) => state.config);
     const allowedScopes: string = useSelector((state: AppState) => state?.auth?.scope);
     const appTitle: string = useSelector((state: AppState) => state?.config?.ui?.appTitle);
+    const UUID: string = useSelector((state: AppState) => state.profile.profileInfo.id);
```
Comment (ashendes) [inline apps/console/src/app.tsx:73]: Lets convert this variable name to lowercase (camel case) as this is not a hard coded final constant.

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx [hunk 3]
```diff
@@ -560,6 +562,8 @@ export const AttributeSettings: FunctionComponent<AttributeSelectionPropsInterfa
     };
 
     const updateValues = () => {
+        eventPublisher.publish("application-user-attribute-click-update-button");
+
```
Comment (ashendes) [inline apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx:566]: Redundant line.

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/script-based-flow.tsx [hunk 4]
```diff
@@ -314,6 +316,10 @@ export const ScriptBasedFlow: FunctionComponent<AdaptiveScriptsPropsInterface> =
      * @param {AdaptiveAuthTemplateInterface} template - Adaptive authentication template.
      */
     const handleTemplateSelection = (template: AdaptiveAuthTemplateInterface) => {
+        eventPublisher.publish("application-sign-in-method-conditional-authentication-template", {
+            "template": template["name"]
+        });
+
```
Comment (ashendes) [inline apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/script-based-flow.tsx:322]: Redundant line.

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/script-based-flow.tsx [hunk 5]
```diff
@@ -335,6 +341,8 @@ export const ScriptBasedFlow: FunctionComponent<AdaptiveScriptsPropsInterface> =
             return;
         }
 
+        eventPublisher.publish("application-sign-in-method-enable-conditional-authentication");
+
```
Comment (ashendes) [inline apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/script-based-flow.tsx:345]: Unnecessary line.

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx [hunk 6]
```diff
@@ -249,6 +252,12 @@ export const AddAuthenticatorModal: FunctionComponent<AddAuthenticatorModalProps
      */
     const handleModalSubmit = (): void => {
 
+        selectedAuthenticators.forEach(element => {
+            eventPublisher.publish("application-sign-in-method-add-new-authenticator", {
+                "authenticator": element["defaultAuthenticator"]["name"]
```
Comment (ashendes) [inline apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx:257]: Null check needed for the primary element?

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx [hunk 7]
```diff
@@ -592,6 +594,8 @@ export const StepBasedFlow: FunctionComponent<AuthenticationFlowPropsInterface>
             options: []
         });
 
+        eventPublisher.publish("application-sign-in-method-click-add-new-step");
+
```
Comment (ashendes) [inline apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx:598]: Unnecessary line.

File: apps/myaccount/src/app.tsx [hunk 8]
```diff
@@ -70,8 +70,11 @@ export const App = (): ReactElement => {
     const config: ConfigReducerStateInterface = useSelector((state: AppState) => state.config);
     const loginInit: boolean = useSelector((state: AppState) => state.authenticationInformation.loginInit);
     const allowedScopes: string = useSelector((state: AppState) => state?.authenticationInformation?.scope);
+    const UUID: string = useSelector((state: AppState) => state.authenticationInformation.profileInfo.id);
```
Comment (ashendes) [inline apps/myaccount/src/app.tsx:73]: Fix naming case here as well.

File: apps/myaccount/src/app.tsx [hunk 9]
```diff
@@ -140,6 +143,17 @@ export const App = (): ReactElement => {
         }
 
     }, [ config?.deployment?.tenant, userName ]);
+    
+    /**
+    * Publish page visit when the UUID is set.
+    */
+    useEffect(() => {
+        if(!UUID) {
```
Comment (ashendes) [inline apps/myaccount/src/app.tsx:151]: Missing space.

File: apps/console/src/features/applications/pages/applications.tsx [hunk 10]
```diff
@@ -355,7 +360,11 @@ const ApplicationsPage: FunctionComponent<ApplicationsPageInterface> = (
                     list={ appList }
                     onApplicationDelete={ handleApplicationDelete }
                     onEmptyListPlaceholderActionClick={
-                        () => history.push(AppConstants.getPaths().get("APPLICATION_TEMPLATES"))
+                        () => {
+                            eventPublisher.publish("application-click-new-application-button");
+
```
Comment (thivi) [inline apps/console/src/features/applications/pages/applications.tsx:365]: Redundant line

File: apps/console/src/features/core/components/header.tsx [hunk 11]
```diff
@@ -235,7 +241,11 @@ export const Header: FunctionComponent<HeaderPropsInterface> = (
                     {
                         icon: "power off",
                         name: t("common:logout"),
-                        onClick: () => history.push(window[ "AppUtils" ].getConfig().routes.logout)
+                        onClick: () => {
+                            eventPublisher.publish("console-click-logout");
+
```
Comment (thivi) [inline apps/console/src/features/core/components/header.tsx:246]: Redundant line

File: apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx (hunk 3 above)
Comment (ThaminduDilshan) [reply apps/console/src/features/applications/components/settings/attribute-management/attribute-settings.tsx:566]: This place has changed with some other code. Hence didn't remove the extra line

File: apps/console/src/app.tsx (hunk 2 above)
Comment (ThaminduDilshan) [reply apps/console/src/app.tsx:73]: Fixed in #2729

File: apps/console/src/app.tsx [hunk 12]
```diff
@@ -143,6 +146,17 @@ export const App: FunctionComponent<{}> = (): ReactElement => {
             search: "?error=" + AppConstants.LOGIN_ERRORS.get("ACCESS_DENIED")
         });
     }, [ config, loginInit ]);
+    
+    /**
+    * Publish page visit when the UUID is set.
+    */
+    useEffect(() => {
+        if(!UUID) {
```
[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply apps/console/src/app.tsx:154]: Fixed in #2729

File: apps/console/src/features/applications/components/edit-application.tsx [hunk 13]
```diff
@@ -407,11 +416,14 @@ export const EditApplication: FunctionComponent<EditApplicationPropsInterface> =
      * Handles the tab change.
      *
      * @param {React.SyntheticEvent} e - Click event.
-     * @param {number} activeIndex - Active tab index.
+     * @param {TabProps} data -
```
[the remark this answers is not in the document]
Comment (ThaminduDilshan) [reply apps/console/src/features/applications/components/edit-application.tsx:419]: Fixed in #2729

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/script-based-flow.tsx (hunk 4 above)
Comment (ThaminduDilshan) [reply apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/script-based-flow.tsx:322]: Fixed in #2729

File: apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/script-based-flow.tsx (hunk 5 above)
Comment (ThaminduDilshan) [reply apps/console/src/features/applications/components/settings/sign-on-methods/script-based-flow/script-based-flow.tsx:345]: Fixed in #2729

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx (hunk 6 above)
Comment (ThaminduDilshan) [reply apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/add-authenticator-modal.tsx:257]: Fixed in #2729

File: apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx (hunk 7 above)
Comment (ThaminduDilshan) [reply apps/console/src/features/applications/components/settings/sign-on-methods/step-based-flow/step-based-flow.tsx:598]: Fixed in #2729

File: apps/myaccount/src/app.tsx (hunk 8 above)
Comment (ThaminduDilshan) [reply apps/myaccount/src/app.tsx:73]: Fixed in #2729

File: apps/myaccount/src/app.tsx (hunk 9 above)
Comment (ThaminduDilshan) [reply apps/myaccount/src/app.tsx:151]: Fixed in #2729

File: apps/console/src/features/applications/pages/applications.tsx (hunk 10 above)
Comment (ThaminduDilshan) [reply apps/console/src/features/applications/pages/applications.tsx:365]: Fixed in #2729

File: apps/console/src/features/core/components/header.tsx (hunk 11 above)
Comment (ThaminduDilshan) [reply apps/console/src/features/core/components/header.tsx:246]: Fixed in #2729

**Pull request merged**